pkgs/by-name/te/teamspeak3/package.nix

```diff
@@ -0,0 +1,143 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  fetchzip,
+  makeDesktopItem,
+  glib,
+  qt5,
+  libsForQt5,
+  perl,
+  libcxx,
+  autoPatchelfHook,
+}:
+
+let
+  arch = "amd64";
+
+  pluginsdk = fetchzip {
+    url = "https://files.teamspeak-services.com/releases/sdk/3.3.1/ts_sdk_3.3.1.zip";
+    hash = "sha256-wx4pBZHpFPoNvEe4xYE80KnXGVda9XcX35ho4R8QxrQ=";
+  };
+in
+
+stdenv.mkDerivation rec {
+  pname = "teamspeak3";
+
+  version = "3.6.2";
+
+  src = fetchurl {
+    url = "https://files.teamspeak-services.com/releases/client/${version}/TeamSpeak3-Client-linux_${arch}-${version}.run";
+    hash = "sha256-WfEQQ4lxoj+QSnAOfdCoEc+Z1Oa5dbo6pFli1DsAZCI=";
+  };
+
+  nativeBuildInputs = [
+    perl # Installer script needs `shasum`
+    qt5.wrapQtAppsHook
+    autoPatchelfHook
+  ];
+
+  buildInputs =
+    [
+      libsForQt5.quazip
+      glib
+      libcxx
+    ]
+    ++ (with qt5; [
+      qtbase
+      qtwebengine
+      qtwebchannel
+      qtwebsockets
+      qtsvg
+    ]);
+
+  # This runs the installer script. If it gets stuck, run it with -x.
+  # If it then gets stuck at something like:
+  #
+  # ++ exec
+  # + PAGER_PATH=
+  #
+  # it's looking for a dependency and didn't find it. Check the script and make
+  # sure the dep is in nativeBuildInputs.
+  unpackPhase = ''
+    runHook preUnpack
+
+    # Run the installer script non-interactively
+    echo -e '\ny' | PAGER=cat sh -e $src
+
+    mv TeamSpeak3-Client-linux_${arch}/* .
```

Comment:
You can just cd into this. It is expected (even for stdenv unpackPhase) that you are in the unpacked content after unpackPhase has completed.

(https://nixos.org/manual/nixpkgs/stable/#var-stdenv-sourceRoot)

Reply:
I can see where you're coming from but I think it's just a lot simpler to do it this way.

Reply:
I think it would be helpful to justify why you think this is simpler - it's possible I'm missing something here.

My argument is that this approach is different to other unpackPhases (and in particular the stock stdenv one), which are usually "take $src file, uncompress it, and change the working directory into the decompressed content". In general, I would consider it simpler to be as similar as possible to other derivations.

Reply:
I don't think it's worth discussing this at this point tbh. This is getting into bikeshedding territory.